packages/orchestration/src/orchestration-client.ts

```diff
@@ -50,6 +50,53 @@ export class OrchestrationClient {
 
     return new OrchestrationResponse(response);
   }
+
+  /**
+   * Executes the orchestration service using a JSON configuration.
+   * @param jsonConfig - The JSON configuration as an input string.
+   * @param prompt - Prompt configuration.
+   * @param requestConfig - Optional request configuration.
+   * @param deploymentConfig - Deployment configuration.
+   * @returns The orchestration response.
+   */
+  /* eslint-disable @typescript-eslint/member-ordering */
+  static async executeFromJson(
+    jsonConfig: string,
+    prompt?: Prompt,
+    requestConfig?: CustomRequestConfig,
+    deploymentConfig?: ResourceGroupConfig
```

Comment:
This might be an API architecture that we need to decide on because currently, the constructor of the OrchestrationClient looks as follows:
```TypeScript
 constructor(
    private config: OrchestrationModuleConfig,
    private deploymentConfig?: ResourceGroupConfig
  ) {}
```
But the whole point of adding the `executeFromJson` method is that users can provide the config as a JSON, and furthermore, it is not mapped to our existing type. 
The workaround I've used is to make the method static and pass `deploymentConfig` as an optional param. 
Do post your opinions/suggestions regarding this.

Reply:
What do you think about adapting the constructor to:

```
constructor(
  private config: OrchestrationModuleConfig | string,
  private deploymentConfig?: ResourceGroupConfig
) {}
```

and then in the `chatCompletion` method check if type is string then parse it into `OrchestrationModuleConfig`.

Reply:
[pp] What about adding a `string` type for `config` in the `OrchestrationClient` constructor?

```ts
constructor(
    private config: OrchestrationModuleConfig | string,
    private deploymentConfig?: ResourceGroupConfig
  ) {}
```

Then in `chatCompletion`:

```ts
let body;
if (typeof this.config === 'string') {
  body = {
    module_configurations: JSON.parse(this.config)
  };
} else {
  body = constructCompletionPostRequest(this.config, prompt);
}
```

So in this case we won't have a separate method `executeFromJson()` but still reuse our `chatCompletion()` method.

Reply:
[q] How about, with the intention of keeping the methods separate (user exp, single responsibility, plus aligns with java), we use a typeguard:

```TypeScript
private isOrchestrationModuleConfig(
    config: string | OrchestrationModuleConfig
  ): config is OrchestrationModuleConfig {
    return typeof config === 'object' && 'templating' in config;
  }
```

and then in both methods, we throw an error (along with a suggestion to use the correct method) if the type doesn't match?

Reply:
From a UX perspective, the current design might be a bit confusing because:

- When passing a TypeScript config object, you instantiate a client and provide the configuration values alongside the deployment configuration.
- However, when using a JSON config, the approach changes entirely—instance properties are replaced by function parameters, and client initialization is no longer required.

Since the ultimate goal in both cases is to use `.chatCompletion()`, it’s not clear why separate methods are necessary.

Reply:
For context: The Java method is called `..FromJsonModuleConfig` mainly so that it's clear what the expected input is. If we want to keep the naming, we could use a static factory method which returns a new `OrchestrationClient`. And in Java we pass the config per-request (and we can't easily have a union type including String), so we need a separate Method there. Of course, this this doesn't apply 1:1 in TS. 

No strong opinion, I think using a type union in the constructor is also fine. In that case we / the user just has to rely more on the TS docs on what to input.

Reply:
Thanks for your suggestions everyone. I ended up going with a type union in the constructor and just using a single `chatCompletion()` method for both cases and handling the construction of the request body accordingly. Also added documentation.